lib/elixir/test/elixir/kernel/diagnostics_test.exs

```diff
@@ -10,6 +10,305 @@ defmodule Kernel.DiagnosticsTest do
+            c,
+            d
+            e )
+          """,
+          MismatchedDelimiterError
+        )
+
+      assert output == expected
+    end
+
+    test "trim inbetween lines if too many" do
+      expected = """
+      ** (MismatchedDelimiterError) mismatched delimiter found on nofile:9:5:
+          error: unexpected token: )
+          HINT: the "[" on line 1 is missing terminator "]"
+          │
+        1 │ [ a,
+          │ └ unclosed delimiter
+          │ ...
+        9 │   i )
+          │     └ mismatched closing delimiter
+          │
+          └─ nofile:9:5\
+      """
+
+      output =
+        capture_raise(
+          """
+          [ a,
+            b,
+            c,
+            d,
+            e,
+            f,
+            g,
+            h
+            i )
+          """,
+          MismatchedDelimiterError
+        )
+
+      assert output == expected
+    end
+
+    test "pads according to line number digits" do
+      expected = """
+      ** (MismatchedDelimiterError) mismatched delimiter found on nofile:13:5:
+          error: unexpected token: )
+          HINT: the "[" on line 1 is missing terminator "]"
+          │
+        1 │ [ a,
+          │ └ unclosed delimiter
+          │ ...
+       13 │   b )
+          │     └ mismatched closing delimiter
+          │
+          └─ nofile:13:5\
+      """
+
+      output =
+        capture_raise(
+          """
+          [ a,
+          #{String.duplicate("\n", 10)}
+            b )
+          """,
+          MismatchedDelimiterError
+        )
+
+      assert output == expected
+
+      expected = """
+      ** (MismatchedDelimiterError) mismatched delimiter found on nofile:403:5:
+           error: unexpected token: )
+           HINT: the "[" on line 1 is missing terminator "]"
+           │
+         1 │ [ a,
+           │ └ unclosed delimiter
+           │ ...
+       403 │   b )
+           │     └ mismatched closing delimiter
+           │
+           └─ nofile:403:5\
+      """
+
+      output =
+        capture_raise(
+          """
+          [ a,
+          #{String.duplicate("\n", 400)}
+            b )
+          """,
+          MismatchedDelimiterError
+        )
+
+      assert output == expected
+
+      expected = """
+      ** (MismatchedDelimiterError) mismatched delimiter found on nofile:107:5:
+           error: unexpected token: )
+           HINT: the "[" on line 99 is missing terminator "]"
+           │
+        99 │ [ a,
+           │ └ unclosed delimiter
+           │ ...
+       107 │   b )
+           │     └ mismatched closing delimiter
+           │
+           └─ nofile:107:5\
+      """
+
+      output =
+        capture_raise(
+          """
+          #{String.duplicate("\n", 97)}
+          [ a,
+          #{String.duplicate("\n", 6)}
+            b )
+          """,
+          MismatchedDelimiterError
+        )
+
+      assert output == expected
+    end
+
+    test "trims lines that are too long (> 60 chars)" do
```

Comment:
What happens when only one of the delimiters is too long ahead?

Reply:
And what happens if it is too many lines and too long lines?

Reply:
The idea is to create a helper function that given a line and the column the error is in, trims it up to a point where `column + delimiter_threshold < 80`, where delimiter threshold is a constant of value around 20 (the length of the delimiter message) so that we always trim it showing the most we can while staying under 80 chars and still showing the delimiters for the line, WDYT?

Reply:
Excellent. Let's make sure we also include the examples from the questions above in the tests. :)

Reply:
If you haven't yet!

Reply:
Hey I've started implementing the line trimming and I've noticed it's really full edge cases and the implementation would be very tricky if we tried to make it have a consistent behavior for all cases. I took a look to see how rust handles it and they don't :laughing: 
[Here's a playground showcasing how it works ](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=f34eeee1c83d0e6dfd54085e178423b4), they simply output the line and don't try to trim it even though it might be a bit long.

After thinking for a while I think that's the best because if we trim it by default users with larger terminals won't be able to see the full lines even though they can, and I think it makes the behavior more consistent with what we have today (currently our `format_snippet` function only trims white spaces in lines and if it's longer than a given threshold.

Let me know your opinions, but I think we can start with simpler code and try to implement this in the future if there's a solid reason to

Reply:
Yeah, I don't think we need to handle this. At best, we trim whitespace.